lazy_tensor_core/lazy_tensor_core/core/optimize.py

```diff
@@ -0,0 +1,90 @@
+from lazy_tensor_core import _LAZYC
+import dataclasses
+from typing import List, Dict, Any, Callable
+import copy
+from torch import fx
+import torch
+
+debug = False
+
+@dataclasses.dataclass
+class GraphInputMatcher:
```

Comment:
it would be good to document this class.  Also change its name? to reflect that it is somewhat specific to Lazy, Fx, Torchscript.  (unless that's not true). Would we make a new matcher for XLA for example?

Reply:
For this class, I think we can reuse it for XLA as long as XLA use the same way as TS to identify graph inputs.